packages/fluentui/react-northstar/src/components/Accordion/Accordion.tsx

```diff
@@ -84,114 +95,108 @@ export interface AccordionProps extends UIComponentProps, ChildrenComponentProps
-    onActiveIndexChange: PropTypes.func,
-    panels: customPropTypes.every([
-      customPropTypes.disallow(['children']),
-      PropTypes.arrayOf(
-        PropTypes.shape({
-          content: customPropTypes.itemShorthand,
-          title: customPropTypes.itemShorthand,
-        }),
-      ),
-    ]),
-
-    renderPanelTitle: PropTypes.func,
-    renderPanelContent: PropTypes.func,
-  };
-
-  static defaultProps = {
-    accessibility: accordionBehavior,
-    as: 'dl',
-  };
-
-  static autoControlledProps = ['activeIndex'];
-
-  static Title = AccordionTitle;
-  static Content = AccordionContent;
-
-  focusHandler: ContainerFocusHandler = null;
-  itemRefs = [];
-  defaultAccordionTitleId = _.uniqueId('accordion-title-');
-  defaultAccordionContentId = _.uniqueId('accordion-content-');
+export const Accordion: React.FC<WithAsProp<AccordionProps>> &
+  FluentComponentStaticProps<AccordionProps> & {
+    Title: typeof AccordionTitle;
+    Content: typeof AccordionContent;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Accordion.displayName, context.telemetry);
+  setStart();
+  const {
+    expanded,
+    exclusive,
+    accessibility,
+    children,
+    className,
+    design,
+    styles,
+    variables,
+    panels,
+    renderPanelContent,
+    renderPanelTitle,
+  } = props;
+  const alwaysActiveIndex = expanded ? 0 : -1;
+
+  const [activeIndex, setActiveIndex] = useAutoControlled({
+    defaultValue: props.defaultActiveIndex,
+    value: props.activeIndex,
+    initialValue: exclusive ? alwaysActiveIndex : [alwaysActiveIndex],
+  });
 
-  actionHandlers = {
+  const actionHandlers = {
     moveNext: e => {
       e.preventDefault();
-      this.focusHandler.moveNext();
+      focusHandler.moveNext();
     },
     movePrevious: e => {
       e.preventDefault();
-      this.focusHandler.movePrevious();
+      focusHandler.movePrevious();
     },
     moveFirst: e => {
       e.preventDefault();
-      this.focusHandler.moveFirst();
+      focusHandler.moveFirst();
     },
     moveLast: e => {
       e.preventDefault();
-      this.focusHandler.moveLast();
+      focusHandler.moveLast();
     },
   };
 
-  constructor(props, context) {
-    super(props, context);
+  const getA11yProps = useAccessibility<AccordionBehaviorProps>(accessibility, {
+    debugName: Accordion.displayName,
+    actionHandlers,
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<AccordionStylesProps>(Accordion.displayName, {
+    className: accordionClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-    this.focusHandler = new ContainerFocusHandler(this.getNavigationItemsSize, this.handleNavigationFocus, true);
-  }
+  const [focusedIndex, setfocusedIndex] = React.useState(0);
```

Comment:
not sure if we want this to be initialised to 0. won't the first item always be focused at first render?

Reply:
You are right. I am changing to use the value of `activeIndex` in the initialization